# Profitable App Profiles for the App Store and Google Play Markets

This is a Data Analysis project aimed to help mobile app developers make data-driven decision with respect to the Applications they choose to build and maintain.

The Goal of this project is to find what type of apps attract more users. The company this project is built for builds free apps that mainly rely on in-app ads, therefore, the revenue generated mainly comes from user count and usage.

# Data Collection (Kaggle Dataset)

# Opening and Exploring Data

We will start by opening the two datasets for the Apple App store and Google Play Store.

In [1]:
from csv import reader

# App Store Dataset
opened_file = open('AppleStore.csv', encoding = 'utf8')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:] #separated header from data for ease.

# Play Store Dataset
opened_file = open('googleplaystore.csv', encoding = 'utf8')
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:] #separated header from data for ease.


For better data readability we will define a function called explore_data().

In [2]:
def explore_data(dataset, start, end, rows_and_columns = False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row
        
    if rows_and_columns:
        print('Number of rows: ', len(dataset))
        print('Number of columns: ', len(dataset[0]))

# Let us check the Apple App store dataset.         
print(ios_header)
print('\n')
explore_data(ios, 0, 4, rows_and_columns = True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of rows:  7197
Number of columns:  16


There are 7197 apps in the Apple App store. some data headers in the column may seem confusing due to proprietary naming. Please check the documentation of the dataset here: [app store documentation](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps)

some column headers of interest:
'track_name', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', and 'prime_genre'

below we check the Google Playstore dataset.

In [3]:
print(android_header)
print('\n')
explore_data(android, 0, 4, rows_and_columns = True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows:  10841
Number of columns:  13


in this dataset there are 10841 android apps and 13 data column headers. We are particularly interested in:

'App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres'

We can note that the Google Dataset has a much more understandable set of column headers. 

for more info please check the Documentation: [play store documentation](https://www.kaggle.com/lava18/google-play-store-apps)

# Data Cleaning

Here we will be cleaning our dataset to remove irrelevant details, detect inaccurate data then correct it, detect duplicates, remove non english apps and remove non-free applications.

Overview:
- Remove/correct wrong data
- Remove duplicates
- Modify data to fit our analysis

# Remove / Correct wrong data

# Google Play Store (android)

Based on a [discussion](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) row with index `10472` has an error. Apparently this data point has a missing category value. 

In [4]:
# Compare the header to the data point
print(android_header)
print('\n')
print(android[10472]) 

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


for purpose the purpose of simplicity, we will delete this data point (row). 

Possible alternative solutions may be to alter the data point.

In [5]:
# Deletion of android[10472]
del android[10472] # Make sure to only run this once.

In [6]:
# Checking new data point at index 10472
print(android[10472])
print('\n')
print('New number of rows: ', len(android))

['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']


New number of rows:  10840


row is succesfully deleted.

# Apple App Store (ios)

Based on [discussion](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/discussion/106176), there seems to be no errors inside the Apple data set. 

# Checking and removing duplicates

# Part 1

# Google Play Store (android)

In [7]:
# Test 'Instagram' (random sample)

for app in android:
    name = app[0] # app name is row index 0.
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


As we can see from our first test above, the android data set does have around 4 entries.

In [8]:
# list for storing name of duplicate apps and unique apps
duplicate_apps = []
unique_apps = []

# loop through the data set to check for the duplicates
for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
        
# Test using print
print('Number of duplicate apps: ', len(duplicate_apps))
print('\n')
print('Examples of duplicate apps: ', duplicate_apps[:15])
# First 15 apps.


Number of duplicate apps:  1181


Examples of duplicate apps:  ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


We can observe that the android dataset has around 1181 cases of duplicate data points.

below that result shows 15 examples of those apps with duplicate instances.

Now knowing this, we must now remove the duplicate rows. But how can we do this? 

At first glance we may think of removing the duplicate data point randomly, but a better solution to this is to keep the data point with the most recent data. 

Besides, in Data Science we are particularly very interested in the relevance of data being used so the more recent the data the better it's quality.

# Apple App Store (ios)

The app store seems to be materially clean in terms of duplicates.

There are no needed actions. If there are duplicates, they are most definitely track_name related. Although, when using the unique id column they are all unique.



# Actually Removing the Duplicates

# Part 2

# Google Play Store (android)

In [9]:
# Show computation for the expected length

print('Expected length: ', len(android) - 1181)

Expected length:  9659


To start removing duplicate entries we will start by creating a dictionary where each key is a unique app name and the value is the highest number of reviews to determine how recent the duplicate's data is.

In [10]:
# Dictionary

reviews_max = {}

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    
    if name not in reviews_max:
        reviews_max[name] = n_reviews

print('max reviews length:',len(reviews_max))

max reviews length: 9659


the supplementary condition is meant to solve issues regarding duplicates that have the same number reviews.
if we do not remove duplicates with same number of reviews this will end up as another set of duplicates.

In [11]:
# lists
android_clean = []
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    # Supplementary condition
    if (n_reviews == reviews_max[name]) and (name not in already_added):
            android_clean.append(app)
            already_added.append(name)
            
# Explore android_clean data set
explore_data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows:  9659
Number of columns:  13


As expected there are now 9659 rows.

# Modelling the Data (Isolating English apps)

# Part One

At our company we use English for the Apps we build. Therefore we would like to analyze apps that are targeted towards a English-speaking audience.

If we check the data sets long enough we will be able to detect apps that are directed towards non-english speaking audiences.

In order to filter the results to an English focused dataset we will use as filters the letters commonly used in the English language such as letters A-Z, numbers 0-9, common punctuation marks and etc.

To achieve this filter we will use the ASCII system.

`if ord(character) <= 127` then this is probably an English-used character.

We will build a function that takes strings as inputs and determines their ASCII number identity, this returns False if greater than 127 otherwise it's clean.

In [12]:
# Function
def eng_checker(name):
    for char in name:
        if ord(char) > 127:
            return False #Probably not English
        
        return True #Probably English
# Testing this function
print(eng_checker('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(eng_checker('Instagram'))
print(eng_checker('Docs To Go™ Free Office Suite'))
print(eng_checker('Instachat 😜'))
print(eng_checker('😜'))

False
True
True
True
False


Function seems to work returns False if non-english and True if english.

# Part Two

There are English apps that contain symmbols that may be considered by the function as a non-english symbol like emojis or certain copyright or trademark symbols.

This can cause data loss. So we will mitigate this issue by limiting detection of less than 3 to qualify as an english app.

In [13]:
# Modification of the function
def eng_checker(name):
    non_ascii = 0
    for char in name:
        if ord(char) > 127:
            non_ascii += 1 
    #print(non_ascii) - debugger    
    if non_ascii > 3: #Probably not English
        return False
    else:
        return True #Probably English

# Testing this new function
print(eng_checker('Docs To Go™ Free Office Suite'))
print(eng_checker('Instachat 😜'))
print(eng_checker('爱奇艺PPS -《欢乐颂2》电视剧热播'))

True
True
False


# Google Playstore (android)

In [14]:
android_eng = []

for app in android_clean:
    name = app[0]
    if eng_checker(name):
        android_eng.append(app)

# Apple App Store (ios)

In [15]:
ios_eng = []

for app in ios:
    name = app[1]
    if eng_checker(name):
        ios_eng.append(app)

In [16]:
# Test. Explore data to check.

# Android
explore_data(android_eng, 0, 3, True)
print('\n')
# Ios
explore_data(ios_eng, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows:  9614
Number of columns:  13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+'

In conclusion of our Data modelling we are now left with 9614 Android Apps and 6183 ios apps.

# Isolating the Free Apps

The company for which the analysis is made specifically said that we create free apps and get revenue only from in-app ads. We will now go through our final cleaning phase which is to remove paid apps.

# Google Play Store (android)

In [17]:
# Retrieving the price header
print(android_header)
print('\n')
print(android_header.index('Price'))

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


7


In [18]:
#final list variable used
android_final = []

for app in android_eng:
    price = app[7]
    if price == '0':
        android_final.append(app)
        

# Apple App Store (ios)

In [19]:
# Retrieving the price header
print(ios_header)
print('\n')
print(ios_header.index('price'))

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


4


In [20]:
#final list variable used
ios_final = []

for app in ios_eng:
    price = app[4]
    if price == '0.0':
        ios_final.append(app)

In [21]:
# Final length of the cleaned data sets
print(len(android_final))
print(len(ios_final))

8864
3222


Our final count for the data sets is now:

- Android: 8864 apps
- Ios: 3222 apps

This concludes the Data Cleaning process.

# Exploratory Data Analysis 

# Part One

# Most Common Apps by Genre

`Free Apps`

Since the profit depends on the number of users using the app (in-app ads). We need to find apps which has the highest number of users on both platforms or the 'most popular' ones. 

Since our end goal is to add the app on both the Google play store and the Apple App Store, we should find app profies that are successful on both app stores.

In [22]:
# Inspect both data sets to identify useful columns

print(android_header)
print('\n')
explore_data(android_final, 0, 3, True)
print('\n')
print(ios_header)
print('\n')
explore_data(ios_final, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows:  8864
Number of columns:  13


['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['28488221

Since our goal is to build a frequency table to check the most common apps by genre the columns of interest to us is the 'prime_genre' column of the App store and the 'Genres' and 'Category' of the Google Play store datasets.

# Part Two

We will first build two functions we can use to analyze the frequency tables.

- One function to generate frequency tables that show percentages

- Another function to display percentages in a descending order

In [23]:
# function 1
def freq_table(dataset, index):
    table = {}
    total = 0
    table_percentages = {}
    
    for row in dataset:
        total += 1
        value = row[index]
        # count each genre's occurences
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    # Occurrences divided by total to get percentage expression
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage
    
    # return total - debugger
    return table_percentages #relative to number of total apps.
    
# function 2 - function 1 built inside this already.
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])
        

In [24]:
# relevant indices
print(ios_header.index('prime_genre'))
print(android_header.index('Genres'))
print(android_header.index('Category'))

11
9
1


# Part Three

In [25]:
# ios prime_genre
display_table(ios_final, 11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


Based on above the most common genre in the Apple App Store is the Games genre accounting for more than half of the total offerings. This is followed by Entertainment at second place. 

Almost majority of the apps are designed for fun and entertainment. Although, fun and entertainment Apps dominate the App store this still does not imply large user counts therefore, we still cannot recommend any solid actions.

In [26]:
# android 'Category'
display_table(android_final, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

The Landscape of the Google Playstore seems different from the Apple App Store. The most common genre seems to be Family. Family is mostly composed of games for kids.

The distribution of genres seems to be much more balanced for the play store as there seems to be a healthy number of both fun apps and practical apps. The App store seems to be much more biased toward fun apps but the playstore is more balanced.

the datapoints below also confirm the analysis. the below datapoints named 'Genres' seems to be a more detailed or broken down version of category although the information we need is more of bigger picture so we will stick to category.

We still can't conclude anything at this point, we must now analyze apps with the most users.

In [27]:
# android 'Genres'
display_table(android_final, 9)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

In [28]:
# Relevant indices

# Android
print('android: ', android_header.index('Installs'))

# ios
print('ios: ', ios_header.index('rating_count_tot'))

android:  5
ios:  5


# The Most Popular Genres for users
For the Android data set we can use the installs column to be able to check the popularity of a genre. But this information cannot be found on the Apple store data.

To fix this we will do a workaround we will use the rating_count_tot, (user ratings). 

# Apple App Store (ios)
Below, we calculate the average number of user ratings per app genre on the App Store:

In [29]:
ios_genres = freq_table(ios_final, 11)

for genre in ios_genres:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[11]
        if genre_app == genre:
            user_ratings = float(app[5])
            total += user_ratings
            len_genre += 1
    # compute average number of user ratings
    ave_user_rating = total / len_genre
    # print app genre and average number of ratings
    print(genre, ':', ave_user_rating)

Productivity : 21028.410714285714
Food & Drink : 33333.92307692308
Shopping : 26919.690476190477
Medical : 612.0
Sports : 23008.898550724636
Health & Fitness : 23298.015384615384
Business : 7491.117647058823
News : 21248.023255813954
Catalogs : 4004.0
Music : 57326.530303030304
Entertainment : 14029.830708661417
Book : 39758.5
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Reference : 74942.11111111111
Finance : 31467.944444444445
Education : 7003.983050847458
Utilities : 18684.456790123455
Travel : 28243.8
Weather : 52279.892857142855
Games : 22788.6696905016
Photo & Video : 28441.54375
Social Networking : 71548.34905660378


Based on the results it seems that Navigation Apps have the highest number of average user ratings.

Let's attempt to check Navigation:

In [30]:
for app in ios_final:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5]) # print name : number of ratings

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


What causes this large average rating is Waze and Google maps which when combined has 499,957 reviews all in all. although there are very few apps here this amount of ratings distributed to this small amount of apps skewed the average rating for Navigation apps to be very large.

But when statistically analyzed there are just 2 very large apps and the rest are quite unpopular, therefore, this skews the data set to the right.

The runner up here is Reference which may be a better choice but we will first analyze the data points

In [31]:
for app in ios_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5]) # print name : number of ratings

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


The data points for Reference apps seems to be similar where when a bell curve is imagined the data seems to also skew to the right. 

The largest data points being the Bible and Dictionary.

Although this seems to be a better option than Navigation apps as this sample has a larger number of observations, therefore, there is more value to this as an option.

We can maybe recommend from this that we build a sort of digitalized book that is turned into an app where features are added into the application. Maybe a Technical handbook, or a digital interactive book. etc.

# Google Play Store (android)

In [32]:
display_table(android_final, 5) # Checking installs column for intervals

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


In [33]:
# Nested loop to check data
# str.replace(old, new) method.

category_android = freq_table(android_final, 1)

for category in category_android:
    total = 0 # store sum of installations for each category
    len_category = 0 # number of apps of each genre
    for app in android_final:
        category_app = app[1]
        if category_app == category:
            num_install = app[5]
            num_install = num_install.replace('+','')
            num_install = num_install.replace(',','')
            total += float(num_install)
            len_category += 1
    ave_num_install = total / len_category
    print(category, ':', ave_num_install)

TRAVEL_AND_LOCAL : 13984077.710144928
FOOD_AND_DRINK : 1924897.7363636363
EVENTS : 253542.22222222222
BEAUTY : 513151.88679245283
HOUSE_AND_HOME : 1331540.5616438356
PERSONALIZATION : 5201482.6122448975
MAPS_AND_NAVIGATION : 4056941.7741935486
ENTERTAINMENT : 11640705.88235294
EDUCATION : 1833495.145631068
FINANCE : 1387692.475609756
TOOLS : 10801391.298666667
VIDEO_PLAYERS : 24727872.452830188
COMMUNICATION : 38456119.167247385
NEWS_AND_MAGAZINES : 9549178.467741935
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
AUTO_AND_VEHICLES : 647317.8170731707
BOOKS_AND_REFERENCE : 8767811.894736841
PRODUCTIVITY : 16787331.344927534
BUSINESS : 1712290.1474201474
FAMILY : 3695641.8198090694
GAME : 15588015.603248259
SHOPPING : 7036877.311557789
SPORTS : 3638640.1428571427
PARENTING : 542603.6206896552
DATING : 854028.8303030303
PHOTOGRAPHY : 17840110.40229885
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
WEATHER : 5074486.197183099
HEALTH_AND_FITNESS : 4188821.985

When looking through the results it seems that communications apps have the highest number of average installs per application.

Lets take a closer look.

In [34]:
for app in android_final:
    if app[1] == 'COMMUNICATION':
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
Messenger for SMS : 10,000,000+
My Tele2 : 5,000,000+
imo beta free calls and text : 100,000,000+
Contacts : 50,000,000+
Call Free – Free Call : 5,000,000+
Web Browser & Explorer : 5,000,000+
Browser 4G : 10,000,000+
MegaFon Dashboard : 10,000,000+
ZenUI Dialer & Contacts : 10,000,000+
Cricket Visual Voicemail : 10,000,000+
TracFone My Account : 1,000,000+
Xperia Link™ : 10,000,000+
TouchPal Keyboard - Fun Emoji & Android Keyboard : 10,000,000+
Skype Lite - Free Video Call & Chat : 5,000,000+
My magenta : 1,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Seznam.cz : 1,000,000+
Antillean Gold Telegram (original version) : 100,000+
AT&T Visual Voicemail : 10,000,000+
GMX Mail : 10,000,000+
Omlet Chat : 10,000,000+
My Vodacom SA : 5,000,000+
Microsoft Edge : 5,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Calls & Text by Mo+ : 5,000,000+
free 

In [35]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Viber Messenger : 500,000,000+


In [40]:
# Test below 100m
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '10,000,000+'
                                      or app[5] == '5,000,000+'):
        print(app[0], ':', app[5])

Messenger for SMS : 10,000,000+
My Tele2 : 5,000,000+
Call Free – Free Call : 5,000,000+
Web Browser & Explorer : 5,000,000+
Browser 4G : 10,000,000+
MegaFon Dashboard : 10,000,000+
ZenUI Dialer & Contacts : 10,000,000+
Cricket Visual Voicemail : 10,000,000+
Xperia Link™ : 10,000,000+
TouchPal Keyboard - Fun Emoji & Android Keyboard : 10,000,000+
Skype Lite - Free Video Call & Chat : 5,000,000+
AT&T Visual Voicemail : 10,000,000+
GMX Mail : 10,000,000+
Omlet Chat : 10,000,000+
My Vodacom SA : 5,000,000+
Microsoft Edge : 5,000,000+
Calls & Text by Mo+ : 5,000,000+
chomp SMS : 10,000,000+
Glide - Video Chat Messenger : 10,000,000+
Text SMS : 10,000,000+
Talkray - Free Calls & Texts : 10,000,000+
GroupMe : 10,000,000+
Contacts+ : 10,000,000+
ExDialer - Dialer & Contacts : 10,000,000+
Full Screen Caller ID : 5,000,000+
Hiya - Caller ID & Block : 10,000,000+
Mr. Number-Block calls & spam : 10,000,000+
CIA - Caller ID & Call Blocker : 5,000,000+
Calls Blacklist - Call Blocker : 10,000,000+
C

In a similar fashion as the Apple datasets we can see that the Communication Apps are skewed to right due to the very famous apps that contain billions of downloads.

Communications apps are mostly made by big companies, therefore we can say that these apps can be quite difficult to compete with.

If we remove these apps from the genre data point the average installs can be ramped further downwards.

Other App genres like Entertainment and Games with tons of Downloads seem to be dominated by large apps built by large corporations as well.

It will be quite hard to compete with apps like that.

Since we want to build an app that is of the same profile for both operating systems. 

We could attempt to analyze thesame choice we did with Apple App store.

In [37]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

In [41]:
# Testing middle areas
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

Based on higher download counts and in the middle areas, there seems to be still a very healthy download count on these types of apps.

trasforming technical books, manuals, famous books or guides to apps can also become quite profitable but the game is now on the features of the application. The more catchy or useful the features the better the competitive advantage.

# Conclusion

Based on our analysis we can conclude that for profitability based on user count a viable app profile for both the Google Playstore and Apple App store, are Books and References related Applications or those applications based on a specific book transformed into an app with features.

Some nice ideas would be a famous book transformed to an app just like the dictionary or the bible as in the above examples.

For other domains, technical manuals, regulatory documents or DIY drawings are very nice ideas also.

The competition now is how much features are integrated into the application, such as a built-in dictionary, maybe a discussion page or perhaps notification-based features and integration with smaller devices (smart watch) etc.